sdks/typescript/src/clients/listeners/durable-listener/pooled-durable-listener-client.ts

```diff
@@ -0,0 +1,227 @@
+          if (emitter) {
+            emitter.responseEmitter.emit('response', event);
+            delete this.subscribers[subId];
+
+            // Remove this subscription from the mapping
+            this.taskSignalKeyToSubscriptionIds[subscriptionKey] =
+              this.taskSignalKeyToSubscriptionIds[subscriptionKey].filter((id) => id !== subId);
+
+            if (this.taskSignalKeyToSubscriptionIds[subscriptionKey].length === 0) {
+              delete this.taskSignalKeyToSubscriptionIds[subscriptionKey];
+            }
+          }
+        }
+      }
+
+      this.client.logger.debug('Durable event listener finished');
+    } catch (e: any) {
+      if (isAbortError(e)) {
+        this.client.logger.debug('Durable event listener aborted');
+        return;
+      }
+      this.client.logger.error(`Error in durable-event-listener: ${e.message}`);
+    } finally {
+      const subscriberCount = Object.keys(this.subscribers).length;
+      if (subscriberCount > 0) {
+        this.client.logger.debug(
+          `Durable event listener loop exited with ${subscriberCount} subscribers`
+        );
+        this.client.logger.debug(`Restarting durable event listener retry ${retryCount + 1}`);
+        this.init(retryCount + 1);
+      }
+    }
+  }
+
+  subscribe(request: { taskId: string; signalKey: string }): DurableEventStreamable {
+    const { taskId, signalKey } = request;
+
+    if (!this.listener) throw new Error('listener not initialized');
+
+    // eslint-disable-next-line no-plusplus
+    const subscriptionId = (this.subscriptionCounter++).toString();
+    const subscriber = new DurableEventStreamable(this.listener, taskId, signalKey);
+
+    this.subscribers[subscriptionId] = subscriber;
+
+    const key = `${taskId}-${signalKey}`;
+    if (!this.taskSignalKeyToSubscriptionIds[key]) {
+      this.taskSignalKeyToSubscriptionIds[key] = [];
+    }
+    this.taskSignalKeyToSubscriptionIds[key].push(subscriptionId);
+
+    this.requestEmitter.emit('subscribe', { taskId, signalKey });
+    return subscriber;
+  }
+
+  async result(request: { taskId: string; signalKey: string }): Promise<DurableEvent> {
+    const subscriber = this.subscribe(request);
+    const event = await subscriber.get();
+    return event;
+  }
+
+  async registerDurableEvent(request: {
+    taskId: string;
+    signalKey: string;
+    sleepConditions: Array<SleepMatchCondition>;
+    userEventConditions: Array<UserEventMatchCondition>;
+  }): Promise<RegisterDurableEventResponse> {
+    const conditions: DurableEventListenerConditions = {
+      sleepConditions: request.sleepConditions,
+      userEventConditions: request.userEventConditions,
+    };
+
+    const registerRequest: RegisterDurableEventRequest = {
+      taskId: request.taskId,
+      signalKey: request.signalKey,
+      conditions,
+    };
+
+    const registerResponse = await this.client.client.registerDurableEvent(registerRequest);
+    return registerResponse;
```

Comment:
nit but an entirely stylistic thing: I've always preferred this:
```suggestion
    return await this.client.client.registerDurableEvent(registerRequest);
```

WDYT?